modules/core/include/opencv2/core.hpp

```diff
@@ -1739,6 +1739,13 @@ should be done separately if needed.
 */
 CV_EXPORTS_W void transpose(InputArray src, OutputArray dst);
 
+/** @brief Transpose for n-dimensional matrices.
+ * @param src input array.
+ * @param order new order of dimensions.
+ * @param dst output array of the same type as src.
+ */
+CV_EXPORTS_W void transpose(const Mat& src, const std::vector<int>& order, OutputArray dst);
```

Comment:
To keep Python bindings compatible, `order` should go last.

If it doesn't not help, then we probably should use `transposeND()` alias or name?

Reply:
Renamed to `transposeND()`.